staticDHCPd/staticdhcpdlib/databases/_caching.py

```diff
@@ -163,15 +163,67 @@ def _cacheMAC(self, mac, definition, chained):
          definition.domain_name, definition.domain_name_servers, definition.ntp_servers,
          definition.lease_time
         )
-        
+
+class MemcachedCache(_DatabaseCache):
+    """
+    A memory database cache using memcache.
+    """
+    _mac_cache = None #: A dictionary of cached MACs
+    _subnet_cache = None #: A dictionary of cached subnet/serial data
+
+    def __init__(self, name, memcached_server, memcached_age_time, chained_cache=None):
+        """
+        Initialises the cache.
+
+        :param basestring name: The name of the cache.
+        :param basestring memcached_server: Address and port to connect to the memcached server.
+        :param basestring memcached_age_time: number of seconds to store items in memcache.
+        :param :class:`_DatabaseCache <_DatabaseCache>` chained_cache: The next
+            node in the chain; None if this is the end.
+        """
+        _DatabaseCache.__init__(self, name, chained_cache=chained_cache)
+        import memcache
+
+        self.mc_client = memcache.Client([memcached_server])
```

Comment:
In the future, we'll want to add for the ability to use multiple memcache servers.